storage/src/main/java/tech/pegasys/teku/storage/server/rocksdb/RocksDbDatabase.java

```diff
@@ -108,6 +118,82 @@ public StorageUpdateResult update(final StorageUpdate event) {
     return doUpdate(event);
   }
 
+  @Override
+  public StorageUpdateResult addEth1Deposit(final DepositWithIndex depositWithIndex) {
+    try (final Updater updater = dao.updater()) {
+      updater.addEth1Deposit(depositWithIndex);
+      updater.commit();
+      return StorageUpdateResult.successfulWithNothingPruned();
+    }
+  }
+
+  @Override
+  public StorageUpdateResult addEth1BlockData(
+      final UnsignedLong timestamp, final Eth1BlockData eth1BlockData) {
+    try (final Updater updater = dao.updater()) {
+      updater.addEth1BlockData(timestamp, eth1BlockData);
+      updater.commit();
+      return StorageUpdateResult.successfulWithNothingPruned();
+    }
+  }
+
+  @Override
+  public StorageUpdateResult pruneEth1Deposits(final UnsignedLong eth1DepositIndex) {
```

Comment:
We can simplify this by creating the `Updater` immediately and deleting entries as we walk through them, rather than creating lists of things to delete first (RocksDb can handle concurrent updates).

And RocksDb keys are ordered so we shouldn't need to walk through the entire stream - as long as we're iterating in the right direction as soon as we find a key that we want to keep, we can stop.

There is a `RocksDB.deleteRange` method which would let us just delete the entire range of keys in one call which would be ideal - the catch is that it's not available via a transaction which makes it a bit harder to expose through the `Updater` interface.  Very tempting to expose it directly on `RocksDbDao` though...  @mbaxter how much would that annoy you? :)

The two deletes wouldn't be atomic together but that's ok as this is data we don't need anymore anyway so even if we crashed before the second one was deleted we'd just prune it on the next go round.

Reply:
No need to return a `StorageUpdateResult` here either.